rasa/utils/tensorflow/metrics.py

```diff
@@ -0,0 +1,282 @@
+import tensorflow as tf
+from tensorflow.keras import backend as K
+from tensorflow.types.experimental import TensorLike
+from typing import Any, Dict, Optional
+
+
+# original code taken from
+# https://github.com/tensorflow/addons/blob/master/tensorflow_addons/metrics/f_scores.py
+# (modified to our neeeds)
```

Comment:
Can we be more specific with the reference that we have taken the code parts from, e.g. using a tag or commit hash? If we ever need to check what we changed ourselves, that would be useful. Same for all the copies.